Userland/Libraries/LibGfx/Range.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2023, Preston Taylor <PrestonLeeTaylor@proton.me>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Format.h>
+
+namespace Gfx {
+
+template<typename T>
+class Range {
```

Comment:
What is this supposed to represent? The commit message just says "Implement a simple Range class" which could mean a lot of things :0

Reply:
I've changed the commit message to be a bit more descriptive.

It's suppose to be a general purpose class like `Rect`, but for this specific PR it's so we can split up the overflow ranges into  x and y coordinates as we have to handle them separately.